dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/JSONObject.java

```diff
@@ -0,0 +1,217 @@
+    public JSONObject addField(String name, String value, boolean jsonEscapeName, boolean jsonEscapeValue, boolean trim) {
+        if (name == null || value == null)
+            return this;
+
+        if (name.equals(OMIT_FIELDS_STRING))
+            return this;
+
+        prepForNewField();
+
+        //append name and value
+        JSONBuilder.append("\"");
+
+        if (jsonEscapeName)
+            jsonEscape3(name);
+        else
+            JSONBuilder.append(name);
+
+        JSONBuilder.append("\":\"");
+
+        if (jsonEscapeValue)
+            jsonEscape3(value);
+        else
+            JSONBuilder.append(value);
+
+        JSONBuilder.append("\"");
+        return this;
+    }
+
+    //for int values
+    public JSONObject addField(String name, int value, boolean jsonEscapeName) {
+        if (name == null)
+            return this;
+
+        if (name.equals(OMIT_FIELDS_STRING))
+            return this;
+
+        prepForNewField();
+
+        //append name and value
+        JSONBuilder.append("\"");
+
+        if (jsonEscapeName)
+            jsonEscape3(name);
+        else
+            JSONBuilder.append(name);
+
+        JSONBuilder.append("\":" + value);
+
+        return this;
+    }
+
+    //for bool values
+    public JSONObject addField(String name, boolean value, boolean jsonEscapeName) {
+        if (name == null)
+            return this;
+
+        if (name.equals(OMIT_FIELDS_STRING))
+            return this;
+
+        prepForNewField();
+
+        //append name and value
+        JSONBuilder.append("\"");
+
+        if (jsonEscapeName)
+            jsonEscape3(name);
+        else
+            JSONBuilder.append(name);
+
+        JSONBuilder.append("\":" + value);
+
+        return this;
+    }
+
+    //for long values
+    public JSONObject addField(String name, long value, boolean jsonEscapeName) {
+        if (name == null)
+            return this;
+
+        if (name.equals(OMIT_FIELDS_STRING))
+            return this;
+
+        prepForNewField();
+
+        //append name and value
+        JSONBuilder.append("\"");
+
+        if (jsonEscapeName)
+            jsonEscape3(name);
+        else
+            JSONBuilder.append(name);
+
+        JSONBuilder.append("\":" + value);
+
+        return this;
+    }
+
+    //for long values
+    public JSONObject addField(String name, float value, boolean jsonEscapeName) {
+        if (name == null)
+            return this;
+
+        if (name.equals(OMIT_FIELDS_STRING))
+            return this;
+
+        prepForNewField();
+
+        //append name and value
+        JSONBuilder.append("\"");
+
+        if (jsonEscapeName)
+            jsonEscape3(name);
+        else
+            JSONBuilder.append(name);
+
+        JSONBuilder.append("\":" + value);
+
+        return this;
+    }
+
+    //get current JSON log as a string
+    public String getJSONString() {
+        return JSONBuilder.toString();
+    }
+
+    public void closeLog() {
```

Comment:
rather than a "closeLog" method, I think this whole class would work a bit better if you follow builder pattern... https://dzone.com/articles/design-patterns-the-builder-pattern.  That would mean you'd make this a build method that would close out the bracket and return a JSON instance.  This class would become a Builder class.